nixos/modules/services/misc/guix/default.nix

```diff
@@ -197,6 +259,11 @@ in
 
   config = lib.mkIf cfg.enable (lib.mkMerge [
     {
+      services.guix.extraArgs =
+        lib.optionals (cfg.substituters.urls != [ ]) [
+          "--substitute-urls='${lib.concatStringsSep " " cfg.substituters.urls}'"
+        ];
+
```

Comment:
I'd suggest adding `--substitute-urls=...` directly to the systemd service script instead of combining it with `services.guix.extraArgs`.